src/Microsoft.OData.Client/Materialization/InstanceAnnotationMaterializationPolicy.cs

```diff
@@ -146,7 +146,7 @@ internal void SetInstanceAnnotations(string navigationPropertyName, ODataResourc
         internal IDictionary<string, object> ConvertToClrInstanceAnnotations(ICollection<ODataInstanceAnnotation> instanceAnnotations)
         {
             var clientInstanceAnnotationValue = new Dictionary<string, object>(StringComparer.Ordinal);
-            if (instanceAnnotations != null || instanceAnnotations.Count > 0)
+            if (instanceAnnotations != null && instanceAnnotations.Count > 0)

```

Comment:
Should we just remove the "&& instanceAnnotations.Count > 0" check completely?

Reply:
Yes; we should be able to get rid of the instanceAnnotations.Count check all together.